streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -351,7 +352,7 @@ public boolean isStartingRunningOrPartitionAssigned() {
     private volatile KafkaFutureImpl<Map<String, KafkaFuture<Uuid>>> producerInstanceIdFuture = new KafkaFutureImpl<>();
     private volatile KafkaFutureImpl<Uuid> threadProducerInstanceIdFuture = new KafkaFutureImpl<>();
 
-    private final PunctuateRatioSlidingWindow punctuateRatioSlidingWindow = new PunctuateRatioSlidingWindow(30000); // window size of 30 seconds
+    private final PunctuateRatioSlidingWindow punctuateRatioSlidingWindow = new PunctuateRatioSlidingWindow(30000, new SystemTime()); // window size of 30 seconds
```

Comment:
We should not create a new `SystemTime` object here, but pass in the `time` instance we already have at hand... Otherwise, we can still not mock it...